validphys2/src/validphys/utils.py

```diff
@@ -5,12 +5,41 @@
 @author: Zahari Kassabov
 """
 import contextlib
+import functools
 import pathlib
 import shutil
 import tempfile
+from typing import Any, Sequence, Mapping, Hashable
 
 import numpy as np
 from validobj import ValidationError, parse_input
+from frozendict import frozendict
+
+
+def make_hashable(obj: Any):
+    # So that we don't infinitely recurse since frozenset and tuples
+    # are Sequences.
```

Comment:
I guess the point of this comment is to justify why do `isinstance(obj, Hashable)` first. Namely, frozenset and tuples are both `Hashable` and `Sequence` (I don't think the second is true for frozenset), and by doing the checks in this order we avoid doing an unnecessary loop. 

Maybe this point can be clarified and generalised a bit
 
```suggestion
    # If the `Sequence` check came before the `Hashable` check we would end up unnecessarily, recursively checking all items in objects that are both `Sequence` and `Hashable`, such as e.g. a `tuple`.  To avoid this, the `Hashable` check comes first. 
```